### ISSUE TITLE
Support for API Management services

### ISSUE THREAD

**Issue by pamelafox** [microsoft/NubesGen#368]
**Is your feature request related to a problem? Please describe.**

I am currently attempting to write Bicep for an API Management service (layered on top of a Function) and it would have sure been handy if NubesGen generated it for me! :)

**Describe the solution you'd like**

I'd like to tick a box to add APIM to my function.

**Describe alternatives you've considered**

I am currently decompiling ARM JSON exported from the Azure Portal into Bicep and working off that. Wish me luck.

**Comment by jdubois** [comment]
Yes it would be awesome to automate this! Most services I see on Azure would benefit from an API Management in front of them, and people don't do it because it's too complicated to set up (like what you currently experience).

**Comment by pamelafox** [comment]
I did manage to get APIM working for my functions app. It was a bit complicated due to the necessity to tie the function app and APIM together, but it works!

Linking here in case it's helpful:
https://github.com/pamelafox/model-to-fastapi-azure-function/blob/main/infra/apimanagement.bicep

Here's a non function app APIM that I found while working on mine:
https://github.com/Azure/teams-event-distributor/blob/main/deployment/module_apim.bicep
